mobile/src/shared/SharedHooks.ts

```diff
@@ -0,0 +1,25 @@
+import type { Dispatch, SetStateAction } from "react";
+
+import { useOutletContext } from "react-router-dom";
+
+import type { NullableUser, NullableUserStateType } from "./SharedTypes";
+import type { UserRepository } from "../user/UserRepository";
+
+interface NullableUserContextType {
+  user: NullableUser;
+  setUser: Dispatch<SetStateAction<NullableUser>>;
+}
+
+export function useUser(): NullableUserStateType {
+  const { user, setUser } = useOutletContext<NullableUserContextType>();
+  return [user, setUser];
+}
+
+interface RemoteUserRepositoryContextType {
+  remoteUserRepository: UserRepository;
+}
+export function useRemoteUserRepository(): UserRepository {
```

Comment:
If the plan is to have both a remote and a local `userRepository` , I would hide the distinction from the consumer of the hook. Rename to `useUserRepository` and implement later the logic to choose between the local and remote one inside the hook? IMO that's the point about having a repository interface ;)